src/m365/spo/commands/file/file-list.ts

```diff
@@ -70,7 +80,8 @@ class SpoFileListCommand extends SpoCommand {
     }
 
     try {
-      const files = await this.getFiles(args.options.folder, args);
+      const folderProperties = await this.getItemCount(args.options.folder, args);
```

Comment:
The way this is currently setup does not do what I'd expect it to do. Let's go through it again:

For every folder do:
- Get ItemCounts
- (when recursive) Get all subfolders (taking into account having to page through all the data) 
- (when recursive) Get all files for those subfolders
- Get all files in the current folder.

In code functions I think that should run as follows:

```typescript
public async commandAction(...) {

  const files = this.getFiles(...);

} 

private async getFiles(...) {
  const folderProperties = await this.getItemCount(...);
  // calculate stuff
  // if recursive:
  const subFolders = await this.getSubfoldersPaged(...); //call the /Folders endpoint as many times as is necessary

  //for each subfolder call the current function
  const files = await this.getFiles(...)

  // Now get all the files in the current folder and call as many times as necessary
  const files = await this.getFilesPaged(...)
}
```

Something like that. Do you see where I'm going with this? For every subfolder you need to do exactly the same. And for /Folders as well as for /Files you need to use $skip and $top to retrieve everything.

Reply:
So it would become something like this:

```typescript
public async commandAction(...) {

  const files = this.getFiles(...);

} 

private async getFiles(...) {
  if (args.options.recursive) {
    const subFolders = await this.getSubfoldersPaged(...); //call the /Folders endpoint as many times as is necessary

     //for each subfolder call the current function
    foreach(var folder in subfolders) {
      const files = await this.getFiles(...)
    } 
  }

  // Now get all the files in the current folder and call as many times as necessary
  const files = await this.getFilesPaged(...)
}

```